datahub-web-react/src/app/permissions/policy/ManagePolicies.tsx

```diff
@@ -289,35 +295,66 @@ export const ManagePolicies = () => {
                 urn: policy?.urn as string, // There must be a focus policy urn.
                 input: toPolicyInput(newPolicy),
             },
-        });
-        message.success(`Successfully ${newState === PolicyState.Active ? 'activated' : 'deactivated'} policy.`);
-        setTimeout(() => {
-            policiesRefetch();
-        }, 3000);
+        }).then(()=>{
+            const updatePolicies= {
+                ...newPolicy,
+                __typename: 'ListPoliciesResult',
+            }
+            updateListPoliciesCache(client,updatePolicies,DEFAULT_PAGE_SIZE);
+            message.success(`Successfully ${newState === PolicyState.Active ? 'activated' : 'deactivated'} policy.`);
+            setTimeout(() => {
+                policiesRefetch();
+            }, 3000);
+        })
+        
         setShowViewPolicyModal(false);
     };
 
     // On Add/Update Policy handler
     const onSavePolicy = (savePolicy: Omit<Policy, 'urn'>) => {
         if (focusPolicyUrn) {
             // If there's an URN associated with the focused policy, then we are editing an existing policy.
-            updatePolicy({ variables: { urn: focusPolicyUrn, input: toPolicyInput(savePolicy) } });
-            analytics.event({
-                type: EventType.UpdatePolicyEvent,
-                policyUrn: focusPolicyUrn,
-            });
+            updatePolicy({ variables: { urn: focusPolicyUrn, input: toPolicyInput(savePolicy) } })
+            .then(()=>{
+                const newPolicy = {
+                    __typename: 'ListPoliciesResult',
```

Comment:
one tiny request: can we please place this logic inside the utis?

i'd rather not have it hanging out in the component

Reply:
create new hook usePolicy added on handler there